hikari/impl/rest.py

```diff
@@ -3300,7 +3331,8 @@ async def create_application_command(
     ) -> commands.ContextMenuCommand:
         ...
 
-    async def create_application_command(
+    @deprecation.deprecated("2.0.0.dev106", "create_slash_command or create_context_menu_command")
+    async def create_application_command(  # type: ignore[override]
         self,
```

Comment:
This isn't quite what I meant here, as in you'll also have to revert this to the point where its has the signature it has on #master and is just an equivalent to create_slash_command (and just calls create_slash_command) since create_context_menu_command doesn't exist on #master.

Since the `type` arg previously didn't exist and guild could be passed positionally this would still be a breaking change in its current form.